packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -80,29 +85,68 @@ class CupertinoNavigationBar extends StatelessWidget implements ObstructingPrefe
     Key key,
     this.leading,
     this.automaticallyImplyLeading = true,
+    this.automaticallyImplyMiddle = true,
+    this.previousPageTitle,
     this.middle,
     this.trailing,
     this.border = _kDefaultNavBarBorder,
     this.backgroundColor = _kDefaultNavBarBackgroundColor,
     this.padding,
     this.actionsForegroundColor = CupertinoColors.activeBlue,
   }) : assert(automaticallyImplyLeading != null),
+       assert(automaticallyImplyMiddle != null),
        super(key: key);
 
   /// Widget to place at the start of the navigation bar. Normally a back button
   /// for a normal page or a cancel button for full page dialogs.
+  ///
+  /// If null and [automaticallyImplyLeading] is true, an appropriate button
+  /// will be automatically created.
   final Widget leading;
 
   /// Controls whether we should try to imply the leading widget if null.
   ///
   /// If true and [leading] is null, automatically try to deduce what the [leading]
   /// widget should be. If [leading] widget is not null, this parameter has no effect.
   ///
+  /// Specifically [CupertinoNavigationBar] would:
```

Comment:
i think maybe "will" would be better than "would" here

Reply:
sg